gamemodes/terrortown/gamemode/client/cl_help_data.lua

```diff
@@ -0,0 +1,254 @@
+
+	self.menuTbl[pos] = table.Copy(HELP_DATA_OBJECT)
+	self.menuTbl[pos].id = id
+
+	return self.menuTbl[pos]
+end
+
+---
+-- Checks if a menu with the given ID is already registered
+-- @param string id The unique menu identifier
+-- @return boolean Returns if the identifier is already used
+-- @realm client
+function HELP_DATA:Exists(id)
+	for i = 1, #self.menuTbl do
+		if self.menuTbl[i].id == id then
+			return true
+		end
+	end
+
+	return false
+end
+
+---
+-- Returns a list of all registered menues that are not admin only and
+-- whose should show function does not return false
+-- @return table A table of all menues
+-- @internal
+-- @realm client
+function HELP_DATA:GetVisibleNormalMenues()
+	local menuTbl = {}
+
+	for i = 1, #self.menuTbl do
+		local menu = self.menuTbl[i]
+
+		-- do not show if it is admin only
+		if menu.adminOnly then continue end
+
+		-- do not show if the shouldShow function returns false
+		if isfunction(menu.shouldShowFn) and menu.shouldShowFn() == false then continue end
+
+		menuTbl[#menuTbl + 1] = menu
+	end
+
+	return menuTbl
+end
+
+---
+-- Returns a list of all registered menues that are admin only as
+-- long as the local player is an admin himself and whose should show
+-- function does not return false
+-- @return table A table of all menues
+-- @internal
+-- @realm client
+function HELP_DATA:GetVisibleAdminMenues()
+	local client = LocalPlayer()
+	local menuTbl = {}
+
+	for i = 1, #self.menuTbl do
+		local menu = self.menuTbl[i]
+
+		-- do not show if it is no admin menu
+		if not menu.adminOnly then continue end
+
+		-- do not show if the player is no admin
+		if not client:IsAdmin() then continue end
+
+		-- do not show if the shouldShow function returns false
+		if isfunction(menu.shouldShowFn) and menu.shouldShowFn() == false then continue end
+
+		menuTbl[#menuTbl + 1] = menu
+	end
+
+	return menuTbl
+end
+
+---
+-- Sets the title of a menu element
+-- @param string title The name, can be a language identifier
+-- @realm client
+function HELP_DATA_OBJECT:SetTitle(title)
+	self.title = title
+end
+
+---
+-- Sets the description of a menu element
+-- @param string description The description, can be a language identifier
+-- @realm client
+function HELP_DATA_OBJECT:SetDescription(description)
+	self.description = description
+end
+
+---
+-- Sets the icon material of a menu element
+-- @param Material iconMat The material of the icon
+-- @realm client
+function HELP_DATA_OBJECT:SetIcon(iconMat)
+	self.iconMat = iconMat
+end
+
+---
+-- Sets the admin only state of a menu element
+-- @param [default=true] boolean adminOnly Set true to show this menu only to admins
+-- @realm client
+function HELP_DATA_OBJECT:AdminOnly(adminOnly)
+	self.adminOnly = adminOnly == nil and true or adminOnly
+end
+
+---
+-- Sets a callback function that can be used to descide at a
+-- later point if a menu should be shown
+-- @param function fn The callback function
+-- @realm client
+function HELP_DATA_OBJECT:ShouldShow(fn)
```

Comment:
Rename the function to indicate, that this will add a callback and not answer the question if this should be shown